### PR TITLE
Feature/auth dry

### PR DESCRIPTION
**Pull request by BradfordMez**
## Description

Me and Bradford realized we could DRY the code by placing the Auth middleware into the root of the endpoint routers instead of repeating it for every endpoint. We also noticed that we didn't have to include the user id in the parameter or body of any request since its given in the profile object attached to the request. We also added a delete endpoint for the bookmarks while we were there.
https://www.loom.com/share/a67a56ab91af4a14b764e315130814b2 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: api/profile/profileRouter.js [hunk 1]
```diff
@@ -218,7 +217,7 @@ router.post('/', authRequired, async (req, res) => {
  *                profile:
```
Comment (ryan-hamblin) [inline api/profile/profileRouter.js:217]: let's be sure when we add code comments to add a bit more description about 'why' the comment is there.

**Review by ryan-hamblin — COMMENTED** [review]
Everything that I saw here is relative to the PR's description.

**Review by ryan-hamblin — COMMENTED** [review]
For future development let's decide whether or not we want to adopt `try catch` over `.then .catch` I saw a couple different instances of both being used in the same file.

**Comment by christina-yun** [comment]
Overall this looks really good and clean! I'd say this is ready to merge.

**Review by christina-yun — APPROVED** [review]
Overall this looks really clean. The comments were cleared up to have been for swagger. I'd say this is ready to merge.

**Review by dominthebox — APPROVED** [review]
Changes look good, happy there was a remove bookmark model added!

**Pull request merged**